### PR TITLE
Minor cleanup in java code

### PR DESCRIPTION
**Pull request by dexamundsen**
No changes to code structure or functionality
Minor cleanup / changes to address warnings and use language features

### CODE REVIEW

File: underlay/src/main/resources/config/criteria/cmssynpuf/criteriaselector/conditions/selector.json [hunk 1]
```diff
@@ -3,7 +3,6 @@
   "displayName": "Condition",
   "isEnabledForCohorts": true,
   "isEnabledForDataFeatureSets": true,
-  "supportsTemporalQueries": true,
```
Comment (dexamundsen) [inline underlay/src/main/resources/config/criteria/cmssynpuf/criteriaselector/conditions/selector.json:6]: duplicated at line 15

File: underlay/src/main/java/bio/terra/tanagra/utils/BigQueryStorageWriter.java [hunk 2]
```diff
@@ -233,8 +233,7 @@ public void onFailure(Throwable throwable) {
           }
         }
 
-        if (throwable instanceof Exceptions.AppendSerializationError) {
-          Exceptions.AppendSerializationError ase = (Exceptions.AppendSerializationError) throwable;
```
Comment (dexamundsen) [inline underlay/src/main/java/bio/terra/tanagra/utils/BigQueryStorageWriter.java:237]: pattern variable used

File: underlay/src/main/java/bio/terra/tanagra/underlay/Underlay.java [hunk 3]
```diff
@@ -479,7 +479,7 @@ private static CriteriaOccurrence fromConfigCriteriaOccurrence(
               entities.stream()
                   .filter(entity -> entity.getName().equals(szOccurrenceEntity.occurrenceEntity))
                   .findFirst()
-                  .get();
```
Comment (dexamundsen) [inline underlay/src/main/java/bio/terra/tanagra/underlay/Underlay.java:482]: Optional.get without use of isPresent. Hence added or else throw

File: underlay/src/main/java/bio/terra/tanagra/query/sql/translator/ApiTranslator.java [hunk 4]
```diff
@@ -255,14 +239,11 @@ default String booleanNotFilterSql(String subFilterSql) {
...
   default String logicalOperatorSql(BooleanAndOrFilter.LogicalOperator operator) {
-    switch (operator) {
-      case AND:
-        return "AND";
-      case OR:
-        return "OR";
-      default:
-        throw new SystemException("Unknown logical operator: " + operator);
-    }
+    return switch (operator) {
```
Comment (dexamundsen) [inline underlay/src/main/java/bio/terra/tanagra/query/sql/translator/ApiTranslator.java:242]: enhanced switch in java17

File: service/src/test/resources/application-test.yaml [hunk 5]
```diff
@@ -3,9 +3,7 @@ tanagra:
     files: [ "cmssynpuf_broad", "aouSR2019q4r4_broad", "sd20230331_verily" ]
 
   feature:
-    artifact-storage-enabled: true
```
Comment (dexamundsen) [inline service/src/test/resources/application-test.yaml:6]: not used and not present in any documentation

File: underlay/src/main/java/bio/terra/tanagra/filterbuilder/impl/core/utils/GroupByCountSchemaUtils.java [hunk 6]
```diff
@@ -89,17 +89,13 @@ public static Map<Entity, List<Attribute>> getGroupByAttributesPerOccurrenceEnti
...
   public static BinaryOperator toBinaryOperator(
       DTUnhintedValue.UnhintedValue.ComparisonOperator comparisonOperator) {
-    switch (comparisonOperator) {
-      case COMPARISON_OPERATOR_EQUAL:
-        return BinaryOperator.EQUALS;
-      case COMPARISON_OPERATOR_LESS_THAN_EQUAL:
-        return BinaryOperator.LESS_THAN_OR_EQUAL;
-      case COMPARISON_OPERATOR_GREATER_THAN_EQUAL:
-        return BinaryOperator.GREATER_THAN_OR_EQUAL;
-      case COMPARISON_OPERATOR_BETWEEN:
```
Comment (marikomedlock) [inline underlay/src/main/java/bio/terra/tanagra/filterbuilder/impl/core/utils/GroupByCountSchemaUtils.java:99]: I think it's useful to detail all the cases, even if they have the same effect.

File: underlay/src/main/resources/config/criteria/cmssynpuf/criteriaselector/conditions/selector.json (hunk 1 above)
Comment (marikomedlock) [inline underlay/src/main/resources/config/criteria/cmssynpuf/criteriaselector/conditions/selector.json:6]: nit: Can you keep this line and remove the other duplicate? I think it's helpful to have all the flags together (e.g. isEnabled).

Comment (dexamundsen) [reply underlay/src/main/resources/config/criteria/cmssynpuf/criteriaselector/conditions/selector.json:6]: done

File: underlay/src/main/java/bio/terra/tanagra/filterbuilder/impl/core/utils/GroupByCountSchemaUtils.java (hunk 6 above)
Comment (dexamundsen) [reply underlay/src/main/java/bio/terra/tanagra/filterbuilder/impl/core/utils/GroupByCountSchemaUtils.java:99]: added it as another case with similar exception

File: underlay/src/main/java/bio/terra/tanagra/api/shared/Literal.java [hunk 7]
```diff
@@ -136,34 +130,29 @@ public boolean isNull() {
   public int compareTo(Literal value) {
     if (isNull && value.isNull()) {
       return 0;
-    } else if (isNull && !value.isNull()) {
+    } else if (isNull) {
       return -1;
-    } else if (!isNull && value.isNull()) {
+    } else if (value.isNull()) {
```
[the remark this answers is not in the document]
Comment (dexamundsen) [reply underlay/src/main/java/bio/terra/tanagra/api/shared/Literal.java]: done

**Pull request merged**